### PR TITLE
ref(alertMessage): converted to ts

### PR DESCRIPTION
**Pull request by priscilawebdev**
**Description:**

Conversion of the component `alertMessage` to typescript

### CODE REVIEW

**Comment by EvanPurkhiser** [comment]
I would flip your semantic commit message, `chore(ts): Convert alertMessage`

**Comment by priscilawebdev** [comment]
@EvanPurkhiser I saw that you often use `chore (ts)`, but I prefer to use chore when I update a dependency

File: src/sentry/static/sentry/app/components/alerts.tsx [hunk 1]
```diff
@@ -7,9 +7,15 @@ import AlertStore from 'app/stores/alertStore';
 import AlertMessage from 'app/components/alertMessage';
 import theme from 'app/utils/theme';
 
-const Alerts = createReactClass({
+type Alert = React.ComponentProps<typeof AlertMessage>['alert'];
+
+type State = {
+  alerts: Array<Alert>;
```
Comment (markstory) [inline src/sentry/static/sentry/app/components/alerts.tsx]: We've generally been using `Alert[]`. I don't think this is a lintable thing though and is more of a nit.

File: src/sentry/static/sentry/app/views/settings/project/projectServiceHookDetails.jsx [hunk 2]
```diff
@@ -138,9 +138,12 @@ export default class ProjectServiceHookDetails extends AsyncView {
           <PanelHeader>{t('Event Validation')}</PanelHeader>
           <PanelBody>
             <PanelAlert type="info" icon="icon-circle-exclamation">
-              Sentry will send the <code>X-ServiceHook-Signature</code> header built using{' '}
-              <code>HMAC(SHA256, [secret], [payload])</code>. You should always verify
-              this signature before trusting the information provided in the webhook.
+              <div>
```
Comment (markstory) [inline src/sentry/static/sentry/app/views/settings/project/projectServiceHookDetails.jsx]: How likely do you think it is that we'll have missed adding a div to an existing alert?

Comment (priscilawebdev) [reply src/sentry/static/sentry/app/views/settings/project/projectServiceHookDetails.jsx]: In order to fix another problem, I've add:

```
display: flex;
flex: 1;
```

and consequently I broke this part of the code..so adding a div fixed the problem...but it seems that `width: 100%` also fixes my problem and we don't need to add extra `divs`..I've updated the code

Comment (markstory) [reply src/sentry/static/sentry/app/views/settings/project/projectServiceHookDetails.jsx]: Should this div remain?

**Review by markstory — APPROVED** [review]

**Comment by priscilawebdev** [comment]
@markstory nope ..already removed it

File: src/sentry/static/sentry/app/components/alertMessage.tsx [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+import Alert from 'app/components/alert';
+import AlertActions from 'app/actions/alertActions';
+import Button from 'app/components/button';
+import {IconClose} from 'app/icons/iconClose';
+import {t} from 'app/locale';
+
+type Alert = {
+  id: string;
+  message: React.ReactNode;
+  type: 'success' | 'error' | 'warning' | 'info';
```
Comment (EvanPurkhiser) [inline src/sentry/static/sentry/app/components/alertMessage.tsx]: This probably could have been `React.ComponentProps<typeof Alert>['type']`

**Pull request merged**